spec/models/user_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validations' do
+    it 'is not valid without a name' do
+      user = User.new(name: nil)
+      expect(user).to_not be_valid
+    end
+
+    it 'is valid with a name' do
+      user = User.new(name: 'John Doe', posts_counter: 3)
+      expect(user).to be_valid
+    end
+
+    it 'is not valid with a negative posts_counter' do
+      user = User.new(name: 'Alice', posts_counter: -3)
+      expect(user).to_not be_valid
+    end
```

Comment:
- [ ] Please make sure to test all your methods in your User model for example the `recent_posts` method.